### PR TITLE
[App] Show logs command to be standalone and re-usable

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR does a bit of cleaning of the logs, so it can be used externally.

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by lantiga — COMMENTED** [review]
Looks great! Only missing some test coverage

File: src/lightning_app/cli/commands/connection.py [hunk 1]
```diff
@@ -129,17 +130,20 @@ def connect(app_name_or_id: str, yes: bool = False):
                         retriever.app_id,
                         target_file=target_file,
                     )
-                    click.echo(f"Storing `{command_name}` under {target_file}")
-                    click.echo(f"You can review all the downloaded commands under {commands_folder} folder.")
+                    pretty_command_name = command_name.replace("_", " ")
```
Comment (ethanwharris) [inline src/lightning_app/cli/commands/connection.py:133]: Lots of repetition of this line for cleaning the command name. Don't think it should be addressed here, just a note for future us haha

File: src/lightning_app/cli/lightning_cli.py [hunk 2]
```diff
@@ -94,97 +97,7 @@ def show() -> None:
...
-    )
-
-    rich_colors = list(ANSI_COLOR_NAMES)
-    colors = {c: rich_colors[i + 1] for i, c in enumerate(components)}
-
-    for log_event in log_reader:
-        date = log_event.timestamp.strftime("%m/%d/%Y %H:%M:%S")
-        color = colors[log_event.component_name]
-        rich.print(f"[{color}]{log_event.component_name}[/{color}] {date} {log_event.message}")
+show.command(logs)
```
Comment (ethanwharris) [inline src/lightning_app/cli/lightning_cli.py:100]: Needs the brackets for compatibility with older click versions:
```suggestion
show.command()(logs)
```

**Review by ethanwharris — APPROVED** [review]
LGTM, small comments

File: src/lightning_app/cli/lightning_cli.py (hunk 2 above)
Comment (tchaton) [reply src/lightning_app/cli/lightning_cli.py:100]: Done !

File: src/lightning_app/cli/commands/connection.py (hunk 1 above)
Comment (tchaton) [reply src/lightning_app/cli/commands/connection.py:133]: Yeah, we should probably clean up a bit the logic around it. I want this change to get into the release. So adding this change too in this PR.

**Review by manskx — APPROVED** [review]

**Pull request merged**